filters/token-amount/filter.py

```diff
@@ -0,0 +1,123 @@
+import operator
+import spacy
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from collections import defaultdict
+from typing import Union
+
+"""
+A filter on if the tokens contain specific keywords a certain number of times.
+"""
+
+
+class TokenAmountFilter(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(
+        self,
+        keywords: Union[list, str] = None,
+        thresholds: Union[list, int] = None,
+        operations: Union[list, str] = None,
+    ):
+        super().__init__()
+        self.max_input_length = self.get_input_length(keywords, thresholds, operations)
+        self.final_operators = self.parse_operator(operations)
+        self.final_keywords = self.convert_scalar_to_list(keywords)
+        self.final_thresholds = self.convert_scalar_to_list(thresholds)
+        self.nlp = spacy.load("en_core_web_sm")
+        self.sanity_check()
+
+    def get_input_length(self, keywords, thresholds, operations):
+        all_inputs = [keywords, thresholds, operations]
+        lengths = defaultdict(int)
+        for key, user_input in zip(
+            ["keywords", "thresholds", "operations"], all_inputs
+        ):
+            if isinstance(user_input, list):
+                lengths[key] = len(user_input)
+            else:
+                lengths[key] = 1
+
+        unique_lengths = set(lengths.values())
+        if unique_lengths == {1}:
+            return 1
+        elif len(unique_lengths) > 2:
+            raise ValueError("One or more lists given with non-matching lengths")
+
+        return max(unique_lengths)
+
+    def convert_scalar_to_list(self, pos_scalar):
+        return (
+            pos_scalar
+            if isinstance(pos_scalar, list)
+            else [pos_scalar for _ in range(self.max_input_length)]
+        )
+
+    def parse_operator(self, op):
+        ops = {
+            ">": operator.gt,
+            "<": operator.lt,
+            ">=": operator.ge,
+            "<=": operator.le,
+            "==": operator.eq,
```

Comment:
I wonder if it would make practical sense to have a `!=` operator here as well for integer values?

Reply:
Hmm, for a condition with `!= 0` we could just use `> 0` instead, and for any other natural number `n` we can always have two conditions where `> n` and `< n`. I feel like having `!=` as an option would just enable more human errors, for example, in the case with `!= 1` one might forget that this doesn't filter sentences with `0` occurrences. Explicitly typing out the two cases might help to prevent that.

Reply:
No, that is a fairly reasonable explanation.